packages/issuer/lib/VcIssuer.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  encodeJsonAsURI,
+  ICredentialIssuerMetadataParametersV1_11,
+  ICredentialSuccessResponse,
+  IIssueCredentialRequest,
+  unsupported_credential_format,
+} from '@sphereon/openid4vci-common'
+
+import { v4 as uuidv4 } from 'uuid'
+
+export class VcIssuer {
+  _issuerMetadata: ICredentialIssuerMetadataParametersV1_11
+  _preAuthorizedCode?: string
+  _userPinRequired?: boolean
+  constructor(issuerMetadata: ICredentialIssuerMetadataParametersV1_11, preAuthorizedCode?: string, userPinRequired?: boolean) {
+    this._issuerMetadata = issuerMetadata
+    this._preAuthorizedCode = preAuthorizedCode
+    this._userPinRequired = userPinRequired
+  }
+
+  public getIssuerMetadata() {
+    return this._issuerMetadata
+  }
+
+  public createCredentialOfferDeeplink(): string {
```

Comment:
I rather would see this moved to a seperate function outside of the issuer class

Reply:
created another ts file: `CredentialOffer.ts` this is the only function in it right now, but I imagine this class will have a couple more